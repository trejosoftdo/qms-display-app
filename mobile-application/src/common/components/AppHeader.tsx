import * as React from 'react';
import { Image, StyleSheet } from 'react-native';
import { Stack } from 'expo-router';
import icon from '../../../assets/header-icon.png';
import AppLightTheme from '../theme';

const styles = StyleSheet.create({
  icon: {
    height: 50,
    width: 119,
  },
});

/**
 * AppHeaderProps defines the props for the App Header Component.
 *
 * @property headerRight - a handler for the header right
 * @property title - header title
 */
type AppHeaderProps = {
  headerRight?: () => React.ReactNode;
};

/**
 * A component represents the application header
 *
 * @param {AppHeaderProps} props - The props for the App Header component.
 */
const AppHeader: React.FC<AppHeaderProps> = (props: AppHeaderProps) => (
  <Stack.Screen
    options={{
      headerTitle: () => <Image source={icon} style={styles.icon} />,
      headerStyle: {
        backgroundColor: AppLightTheme.colors.primary,
      },
      headerTitleStyle: {
        fontFamily: AppLightTheme.fonts.headlineLarge.fontFamily,
      },
      headerBackTitleStyle: {
        fontFamily: AppLightTheme.fonts.headlineLarge.fontFamily,
      },
      headerTintColor: AppLightTheme.colors.background,
      headerRight: props.headerRight,
    }}
  />
);


export default AppHeader;