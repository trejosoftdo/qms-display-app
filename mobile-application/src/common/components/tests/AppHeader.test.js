import React from 'react';
import AppHeader, { headerTitle } from '../AppHeader';


describe('AppHeader component', () => {
  const mockHeaderRight = <div>Header Right</div>;
  const props = {
    title: 'Test title',
    headerRight: mockHeaderRight,
  };

  it('renders the component with the provided props as expected', () => {
    expect(renderShallow(<AppHeader {...props} />)).toMatchSnapshot();
  });

  it('renders the header title component as expected', () => {
    expect(renderShallow(headerTitle())).toMatchSnapshot();
  });
});