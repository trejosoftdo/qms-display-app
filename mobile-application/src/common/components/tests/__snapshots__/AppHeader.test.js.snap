// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader component renders the component with the provided props as expected 1`] = `
<Screen
  options={
    {
      "headerBackTitleStyle": {
        "fontFamily": "Montserrat-Regular",
      },
      "headerRight": <div>
        Header Right
      </div>,
      "headerStyle": {
        "backgroundColor": "#2929cc",
      },
      "headerTintColor": "#ffffff",
      "headerTitle": [Function],
      "headerTitleStyle": {
        "fontFamily": "Montserrat-Regular",
      },
    }
  }
/>
`;

exports[`AppHeader component renders the header title component as expected 1`] = `
<Image
  source={1}
  style={
    {
      "height": 50,
      "width": 119,
    }
  }
/>
`;
